types/openfin/v49/index.d.ts

```diff
@@ -0,0 +1,2034 @@
+         *
+         * Creates a new OpenFin Window
+         *
+         * A basic window that wraps a native HTML window. Provides more fine-grained control over the window state such as the ability to minimize,
+         * maximize, restore, etc. By default a window does not show upon instantiation; instead the window's show() method must be invoked manually.
+         * The new window appears in the same process as the parent window.
+         * @param options - The options of the window
+         * @param [callback] - Called if the window creation was successful
+         * @param [callback.successObj] - httpResponseCode
+         */
+        new (
+            options: WindowOption,
+            callback?: (successObj: { httpResponseCode: number }) => void,
+            errorCallback?: (reason: string, errorObj: NetworkErrorInfo) => void): OpenFinWindow;
+        /**
+         * Returns an instance of the current window.
+         * @returns Current window
+         */
+        getCurrent(): OpenFinWindow;
+        /**
+         * Returns a Window object that wraps an existing window.
+         */
+        wrap(appUuid: string, windowName: string): OpenFinWindow;
+    }
+
+    /**
+     * Window
+     * A basic window that wraps a native HTML window. Provides more fine-grained control over the window state such as the ability to minimize,
+     * maximize, restore, etc. By default a window does not show upon instantiation; instead the window's show() method must be invoked manually.
+     * The new window appears in the same process as the parent window.
+     */
+    interface OpenFinWindow {
+        /**
+         * uuid of the application that the window belongs to.
+         */
+        uuid: string;
+        /**
+         * Name of window
+         */
+        name: string;
+        /**
+         * Returns the native JavaScript "window" object for the window. This method can only be used by the parent application or the window itself,
+         * otherwise it will return undefined. The same Single-Origin-Policy (SOP) rules apply for child windows created by window.open(url) in that the
+         * contents of the window object are only accessible if the URL has the same origin as the invoking window. See example below.
+         * Also, will not work with fin.desktop.Window objects created with fin.desktop.Window.wrap().
+         * @returns Native window
+         */
+        getNativeWindow(): Window;
+        /**
+         * Gets the parent application.
+         * @returns Parent application
+         */
+        getParentApplication(): OpenFinApplication;
+        /**
+         * Gets the parent window.
+         */
+        getParentWindow(): OpenFinWindow;
+        /**
+         * Registers an event listener on the specified event.
+         */
+        addEventListener(
+            type: OpenFinWindowEventType,
+            listener: (event: WindowBaseEvent
+					   | WindowAuthRequestedEvent
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/openfin/v49/index.d.ts;
  Line: 838;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.